tests/python/pants_test/backend/jvm/tasks/jvm_compile/java/test_cache_compile_integration.py

```diff
@@ -216,7 +213,17 @@ def complete_config(config):
 
         # Compile, and confirm that we have the right count of artifacts.
         self.run_compile(spec, complete_config(c.config), workdir)
-        self.assertEquals(c.artifact_count, len(os.listdir(artifact_dir)))
+
+        new_artifact_dir = self.get_cache_subdir(cache_dir)
+        if artifact_dir:
```

Comment:
Won't this always be false, since this is initialized to `None`?

Reply:
I was thinking it would begin as `None`, then be assigned within the first iteration and never re-assigned. However, after thinking over your comment, I realized that the `self.assertEqual(new_artifact_dir, artifact_dir)` doesn't need to be tested and have removed it in 21207ec.